velox/expression/PrestoCastHooks.cpp

```diff
@@ -65,6 +70,50 @@ Expected<int32_t> PrestoCastHooks::castStringToDate(
   return util::castFromDateString(dateString, util::ParseMode::kStandardCast);
 }
 
+namespace {
+
+using double_conversion::StringToDoubleConverter;
+
+template <typename T>
+Expected<T> doCastToFloatingPoint(const StringView& data) {
+  static const T kNan = std::numeric_limits<T>::quiet_NaN();
+  static StringToDoubleConverter stringToDoubleConverter{
+      StringToDoubleConverter::ALLOW_TRAILING_SPACES,
+      /*empty_string_value*/ kNan,
+      /*junk_string_value*/ kNan,
+      "Infinity",
+      "NaN"};
+  int processedCharactersCount;
+  T result;
+  auto* begin = std::find_if_not(data.begin(), data.end(), [](char c) {
+    return (c >= '\t' && c <= '\r') || c == ' ';
+  });
+  auto length = data.end() - begin;
```

Comment:
Should we check if length == 0?

Reply:
The current code handles input strings with only space properly. I added a benchmark locally with this kind of input and I don't see a performance difference before and after adding a check of if (length == 0). So I think we don't need to add it here.

Reply:
Got it. Thank you for clarifying,